### DISCUSSION TITLE
Github Copilot LLM model feels outdated

### DISCUSSION THREAD

**Discussion by NGanip** [community/community#137594] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

will github copilot upgrade their LLM models? it feels outdated, after using another AI chatbot like Claude, ChatGPT, and others. the way github copilot chat is weird compared to others, or is it just me?

One of the main use of github copilot for me is to explain stuff, and what bothers me is the style of the language, it feels weird, stiffly maybe? I can't really explain it, sorry.

**Comment by meyerluanna** [comment]
Hello,

Thank you for your feedback on GitHub Copilot. Your experience highlights the importance of continuously updating AI chatbots to ensure they remain effective, especially when they're used to explain complex concepts. While there's no specific update on Copilot’s LLM models yet, GitHub values user feedback for guiding improvements.

Please keep an eye on GitHub’s announcements for any updates and consider sharing your insights directly through GitHub’s feedback channels.

Best regards,
Meyer Luanna,

**Reply by PGTBoos** [reply]
Thanks for the feedback response to GitHub Copilot! I guess it's like raising a toddler—you drop it off at kindergarten, get busy with life, and when you come back, you're wondering if someone forgot to pick it up! 😂 It's still figuring out its words in the most endearing way. We're all excited to see it grow up and start forming full sentences like the other AI chatbots. Here's to hoping it graduates from the toddler phase soon and surprises us all!

Cheers, from ChatGPT

**Comment by maxpavlov** [comment]
Yes, current paid version of copilot can't even distinguish from a proper scss and sass syntax, making a factually incorrect suggestions. Basically anyone using free version of Claude gets a better coding suggestions then copilot could currently provide. Let alone the tiny context window which co-pilot has (it looks into one file in your project and can't provide suggestions relavant to the open project as a whole). Hope this will get improved soon as the product is positioned to be leading the co-pilot pack.

**Comment by NGanip** [comment]
welp, I guess i just cancel my subscription and change into another one

**Reply by maxpavlov** [reply]
I'm not even sure there are tools superior to co-pilot. Does Code Whisperer or codenium support chat? Co-pilot is one of the few AI-based tools that does have a healthy product-market fit, yet it's left rather unattended by its product team. Ideally, co-pilot should announce a backing LLM upgrade every time OpenAI releases a new major model.

**Comment by Brandon-Meneses** [comment]
I understand your point, and I’ve had similar experiences. GitHub Copilot is fantastic for code suggestions and auto-completions, but when it comes to explaining concepts or engaging in more conversational interactions, it can feel a bit limited compared to other LLM-based chatbots like ChatGPT or Claude. This is likely due to the fact that GitHub Copilot was primarily designed for in-line coding assistance, so its conversational capabilities might not be as polished as those of general-purpose AI chatbots.

It would be interesting to see an upgrade to the underlying LLM models for GitHub Copilot, especially for the chat feature, as you mentioned. Having a more natural, fluid conversational style could definitely improve the experience, particularly when you're looking for more in-depth explanations or help with complex topics.

I think an upgrade to a more recent LLM, or tuning the model to be more conversational, would make Copilot even more useful beyond just coding. Let's hope this is something the team is considering for future updates!

**Reply by maxpavlov** [reply]
Completely agree. Co-pilot has been immensely helpful with inline suggestions. You type what you want to achieve in a comment and when you move to the next line the suggested code is almost always exactly what I need. But we want to rely on the Co-pilot to help with introducing conceptual stuff into the project architecture (application architecture), not just be a precise itellisense.

**Reply by maxpavlov** [reply]
Well this didn't age well. Since the update that allowed user to choose the responding model we are back on track with good programming suggestions (if o1-preview is used :) )
